doc/api/packages.md

```diff
@@ -0,0 +1,830 @@
+where the nearest parent `package.json` file contains no top-level `"type"`
+field, or string input without the flag `--input-type`. This behavior is to
+preserve backward compatibility. However, now that Node.js supports both
+CommonJS and ES modules, it is best to be explicit whenever possible. Node.js
+will treat the following as CommonJS when passed to `node` as the initial input,
+or when referenced by `import` statements within ES module code:
+
+* Files ending in `.cjs`.
+
+* Files ending in `.js` when the nearest parent `package.json` file contains a
+  top-level field `"type"` with a value of `"commonjs"`.
+
+* Strings passed in as an argument to `--eval` or `--print`, or piped to `node`
+  via `STDIN`, with the flag `--input-type=commonjs`.
+
+### `package.json` `"type"` field
+
+Files ending with `.js` will be loaded as [ES modules][] when the nearest parent
+`package.json` file contains a top-level field `"type"` with a value of
+`"module"`.
+
+The nearest parent `package.json` is defined as the first `package.json` found
+when searching in the current folder, that folder’s parent, and so on up
+until the root of the volume is reached.
+
+<!-- eslint-skip -->
+```js
+// package.json
+{
+  "type": "module"
+}
+```
+
+```bash
+# In same folder as preceding package.json
+node my-app.js # Runs as ES module
+```
+
+If the nearest parent `package.json` lacks a `"type"` field, or contains
+`"type": "commonjs"`, `.js` files are treated as [CommonJS][]. If the volume
+root is reached and no `package.json` is found, Node.js defers to the default, a
+`package.json` with no `"type"` field.
+
+`import` statements of `.js` files are treated as ES modules if the nearest
+parent `package.json` contains `"type": "module"`.
+
+```js
+// my-app.js, part of the same example as above
+import './startup.js'; // Loaded as ES module because of package.json
+```
+
+Package authors should include the `"type"` field, even in packages where all
+sources are [CommonJS][]. Being explicit about the `type` of the package will
+future-proof the package in case the default type of Node.js ever changes, and
+it will also make things easier for build tools and loaders to determine how the
+files in the package should be interpreted.
+
+Regardless of the value of the `"type"` field, `.mjs` files are always treated
+as ES modules and `.cjs` files are always treated as [CommonJS][].
+
+### Package scope and file extensions
+
+A folder containing a `package.json` file, and all subfolders below that folder
+until the next folder containing another `package.json`, are a
+_package scope_. The `"type"` field defines how to treat `.js` files
+within the package scope. Every package in a
+project’s `node_modules` folder contains its own `package.json` file, so each
+project’s dependencies have their own package scopes. If a `package.json` file
+does not have a `"type"` field, the default `"type"` is `"commonjs"`.
```

Comment:
Two issues here: (1) it's not strictly correct that everything in node_modules necessarily has a package.json, (2) the type boundary does stop on node_modules which is important information too here.

Suggested rewording:

```suggestion
_package scope_. Package scopes do not carry through `node_modules` folders. The
`"type"` the module format of `.js` files within the package scope. If a
`package.json` file does not have a `"type"` field, the default is `"commonjs"`.
```